CMakeLists.txt

```diff
@@ -109,77 +112,74 @@ set(MWENGINE_PROCESSORS ${CPP_SRC}/processors/bitcrusher.cpp
-                         ${JNI_SOURCES})
+# build the mock driver into the library when unit test execution is enabled
 
-    # link all the NDK libraries used by the project
-
-    target_link_libraries(mwengine android atomic log OpenSLES aaudio)
+if (${RUN_TESTS} MATCHES "true")
+    SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -D=MOCK_ENGINE")
+    set(MWENGINE_CORE_SOURCES ${MWENGINE_CORE_SOURCES}
+                              ${CPP_SRC}/drivers/mock_io.cpp)
+endif()
 
-    #####################################
-    ## SWIG JNI to Java class wrapping ##
-    #####################################
+add_library(mwengine SHARED
+                     ${MWENGINE_CORE_SOURCES}
+                     ${MWENGINE_SYNTH_SOURCES}
+                     ${MWENGINE_PROCESSORS}
+                     ${MWENGINE_MIKROWAVE_SOURCES}
+                     ${JNI_SOURCES})
 
-    find_package(SWIG REQUIRED)
-    include(${SWIG_USE_FILE})
+# link all the NDK libraries used by the project
 
-    # Ensure file recognized as C++ (otherwise, exported as C file)
+target_link_libraries(mwengine android atomic log OpenSLES aaudio)
 
-    set_property(SOURCE ${CPP_SRC}/mwengine.i PROPERTY CPLUSPLUS ON)
+#####################################
+## SWIG JNI to Java class wrapping ##
+#####################################
 
-    # Setup SWIG flags and locations
+find_package(SWIG REQUIRED)
+include(${SWIG_USE_FILE})
 
-    set(CMAKE_SWIG_FLAGS -DNO_RTIO -c++ -lcarrays.i -package nl.igorski.mwengine.core)
-    set(CMAKE_SWIG_OUTDIR ${JAVA_SRC}/nl/igorski/mwengine/core)
+set_property(SOURCE ${CPP_SRC}/mwengine.i PROPERTY CPLUSPLUS ON)
 
-    # Export a wrapper file to Java, and link with the created C++ library
+set(CMAKE_SWIG_FLAGS -DNO_RTIO -c++ -lcarrays.i -package nl.igorski.mwengine.core)
+set(CMAKE_SWIG_OUTDIR ${JAVA_SRC}/nl/igorski/mwengine/core)
 
-    swig_add_module(mwengine_wrapped java ${CPP_SRC}/mwengine.i)
-    swig_link_libraries(mwengine_wrapped mwengine)
+# Export a wrapper file to Java, and link with the created C++ library
 
-else()
+swig_add_module(mwengine_wrapped java ${CPP_SRC}/mwengine.i)
+swig_link_libraries(mwengine_wrapped mwengine)
 
-    ################
-    ## UNIT TESTS ##
-    ################
+################
+## UNIT TESTS ##
+################
 
-    # because the use of the mock engine needs to be defined during compilation
-    # of the mwengine library, this is a separate configuration requiring a complete
-    # rebuild of the engine. TODO: https://github.com/igorski/MWEngine/issues/106
+if (${RUN_TESTS} MATCHES "true")
 
-    SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -D=MOCK_ENGINE")
+    # only run for a single architecture, assume x86_64 TODO: specify via local.properties?
 
-    add_library(mwengine STATIC
-                         ${MWENGINE_CORE_SOURCES}
-                         ${MWENGINE_SYNTH_SOURCES}
-                         ${MWENGINE_PROCESSORS})
+    if (${ANDROID_ABI} MATCHES "x86_64")
```

Comment:
Probably a good idea to make this configurable from _local.properties_ as well to allow for CI/CD pipeline flexibility.